test/tests/menu-button_links.js

```diff
@@ -270,13 +258,24 @@ ariaTest('"enter" on role="menuitem"', exampleFile, 'menu-enter', async (t) => {
     await t.context.session.get(t.context.url);
     const item = (await t.context.queryElements(t, ex.menuitemSelector))[index];
 
+    // Update url to remove external reference for dependable testing
+    const newUrl = t.context.url + '#test-url-change';
+    await t.context.session.executeScript(
+      function () {
+        let [selector, index, newUrl] = arguments;
+        document.querySelectorAll(selector)[index].href = newUrl;
+      },
+      ex.menuitemSelector,
+      index,
+      newUrl
+    );
```

Comment:
My suggestion above could be used to replace this by providing a different selector: 

```js
onKeydownGoToLink(t, `${ex.menuitemSelector}:nth-of-type(${index}n)`, newUrl)
```

For a quick example: 
```js
document.querySelector('body:nth-of-type(1n)') === document.querySelectorAll('body')[0]
```